modules/gapi/samples/onevpl_source.cpp

```diff
@@ -0,0 +1,292 @@
+        const int outWidth = 2048;
+        cv::Mat maskImg(outHeight, outWidth, CV_8UC3);
+        const int* const classes = detected_classes.ptr<int>();
+        for (int rowId = 0; rowId < outHeight; ++rowId) {
+            for (int colId = 0; colId < outWidth; ++colId) {
+                size_t classId = static_cast<size_t>(classes[rowId * outWidth + colId]);
+                maskImg.at<cv::Vec3b>(rowId, colId) =
+                    classId < colors.size()
+                        ? colors[classId]
+                        : cv::Vec3b{0, 0, 0}; // sample detects 20 classes
+            }
+        }
+        cv::resize(maskImg, out, in.size());
+        const float blending = 0.3f;
+        out = in * blending + out * (1 - blending);
+*/
+    out = in;
+    }
+};
+} // namespace custom
+
+namespace detail {
+typename cv::gapi::wip::CFGParams::value_type create_from_string(const std::string &line);
+}
+
+int main(int argc, char *argv[]) {
+
+    cv::CommandLineParser cmd(argc, argv, keys);
+    cmd.about(about);
+    if (cmd.has("help")) {
+        cmd.printMessage();
+        return 0;
+    }
+
+    // get file name
+    std::string file_path = cmd.get<std::string>("input");
+    const std::string output = cmd.get<std::string>("output");
+    const auto model_path    = cmd.get<std::string>("ssm");
+    const auto weights_path  = get_weights_path(model_path);
+
+    // get VPL params from cmd
+    int param_index = 1;
+    cv::gapi::wip::CFGParams source_cfgs;
+    try {
+        do {
+            std::string line = cmd.get<std::string>("cfg_param_" + std::to_string(param_index));
+            if (line.empty()) {
+                break;
+            }
+            source_cfgs.insert(detail::create_from_string(line));
+            param_index++;
+        } while(true);
+    } catch (const std::exception& ex) {
+        std::cerr << "Invalid cfg parameter: " << ex.what() << std::endl;
+        return -1;
+    }
+
+    const auto device        = "CPU";
+    G_API_NET(SemSegmNet, <cv::GMat(cv::GMat)>, "semantic-segmentation");
+    const auto net = cv::gapi::ie::Params<SemSegmNet> {
+        model_path, weights_path, device
+    };
+    const auto kernels = cv::gapi::kernels<custom::OCVPostProcessing>();
+    const auto networks = cv::gapi::networks(net);
+
+    // Now build the graph
+    cv::GFrame in;
+    cv::GMat detected_classes = cv::gapi::infer<SemSegmNet>(in);
+    cv::GFrame out = custom::PostProcessing::on(in, detected_classes);
+
+    cv::GStreamingCompiled pipeline = cv::GComputation(cv::GIn(in), cv::GOut(out))
+        .compileStreaming(cv::compile_args(kernels, networks));
+
+    // Create source
+    cv::Ptr<cv::gapi::wip::IStreamSource> cap;
+    try {
+        cap = cv::gapi::wip::make_vpl_src(file_path, source_cfgs);
+        std::cout << "CAP desr: " << cap->descr_of() << std::endl;
+    } catch (const std::exception& ex) {
+        std::cerr << "Cannot create source: " << ex.what() << std::endl;
+        return -1;
```

Comment:
Do you need `try/catch` logic here?

Reply:
I'm sure yes, because "ex.what" returns hierarchical troubleshooting message: param/lib/api/errcode and so on